monkeytype/typing.py

```diff
@@ -91,11 +137,8 @@ def shrink_types(types):
     if len(types) == 0:
         return Any
     if all(is_typed_dict(typ) for typ in types):
-        return shrink_typed_dict_types(types)
-    all_dict_types = tuple(typed_dict_to_dict(typ)
-                           if is_typed_dict(typ)
-                           else typ
-                           for typ in types)
+        return shrink_typed_dict_types(types, max_typed_dict_size)
+    all_dict_types = tuple(RewriteTypedDictToDict().rewrite(typ) for typ in types)
```

Comment:
It seems like `RewriteTypedDictToDict` will do a recursive rewrite of all nested `TypedDict` to `Dict` also. But is that actually correct here? I don't think it is. All we've determined is that at the top level we don't have all typed dicts, so we need to convert them to regular dict. But why does that imply that any value type that is a `TypedDict` should also be converted to a regular dict?

Reply:
I think that `shrink_types(AnonymousTypedDict({'a': AnonymousTypedDict({'b': int})}), Dict[str, int])` should be `Union[Dict[str, Dict[str, int]], Dict[str, int]]` rather than `Union[Dict[str, AnonymousTypedDict({'b': int})], Dict[str, int]]`. 

Given that a top-level mismatch probably means that the nested types won't match, it seems a bit weird to emit a TypedDict class for just one part of the Union. (This holds only for our anonymous TypedDicts. A user-defined TypedDict will be treated like any other type.)